src/OrderPage.tsx

```diff
@@ -46,24 +74,50 @@ function OrderPage() {
     },
   });
 
+  useEffect(() => {
+    if (user?.name) {
+      setValue("sender", user.name);
+    }
+  }, [user?.name, setValue]);
+
   const totalQuantity = watch("receivers").reduce((sum, r) => sum + (r.quantity || 0), 0);
 
-  const totalPrice = product?.price.sellingPrice * totalQuantity;
+  const totalPrice = product ? product.price * totalQuantity : 0;
 
-  const onSubmit = (data: FormData) => {
-    const receiverLines = data.receivers.map((r) => `- ${r.name} / ${r.phone} / 수량: ${r.quantity}`).join("\n");
-    const message = 
-`🎁 ${rank}등 상품 주문 완료!
-보낸 사람: ${data.sender}
-메시지: ${data.message}
-받는 사람 목록:
-${receiverLines}`;
+  const onSubmit = async (data: FormData) => {
+    if (!product) return;
+    try {
+      await orderProduct({
+        productId: product.id,
+        message: data.message,
+        messageCardId: String(selectedCardId),
+        ordererName: data.sender,
+        receivers: data.receivers.map((r) => ({
+          name: r.name,
+          phoneNumber: r.phone,
+          quantity: r.quantity,
+        })),
+      }, user?.authToken || "");
+      const message =
+        `주문이 완료되었습니다.
+상품명: ${product?.name}
+구매 수량: ${totalQuantity}
+발신자 이름: ${data.sender}
+메시지: ${data.message}`;
 
-    alert(message);
-    navigate("/");
+      alert(message);
+      navigate("/");
+    } catch (error) {
+      if (error instanceof AxiosError && error.response?.status === 401) {
```

Comment:
```suggestion
      if (error instanceof AxiosError && error.response?.status === HttpStatusCode.Unauthorized) {
      // import { HttpStatusCode } from "axios";
```

응답 코드는 수천만번 사용됐을 상수이기 때문에.. axios에 정의돼있습니다!